go.mod

```diff
@@ -15,3 +15,9 @@ require (
 	k8s.io/apimachinery v0.21.2
 	sigs.k8s.io/yaml v1.2.0
 )
+
+require (
```

Comment:
why are there 2 require statements?

Reply:
This is because Go 1.17 now has pruned module graph. You can see the full details in the changelog here: https://golang.org/doc/go1.17#go-command

> For the go command to correctly resolve transitive imports using the pruned module graph, the go.mod file for each module needs to include more detail about the transitive dependencies relevant to that module. If a module specifies go 1.17 or higher in its go.mod file, **its go.mod file now contains an explicit require directive for every module that provides a transitively-imported package**. (In previous versions, the go.mod file typically only included explicit requirements for directly-imported packages.)

Reply:
lovely, thanks TIL